devtools/gradle/gradle-model/src/main/java/io/quarkus/gradle/tooling/ToolingUtils.java

```diff
@@ -37,14 +39,21 @@ public static boolean isEnforcedPlatform(ModuleDependency module) {
                 || Category.REGULAR_PLATFORM.equals(category.getName()));
     }
 
-    public static IncludedBuild includedBuild(final Project project, final String projectName) {
+    public static IncludedBuild includedBuild(final Project project,
+            final ProjectComponentIdentifier projectComponentIdentifier) {
         try {
-            return project.getGradle().includedBuild(projectName);
-        } catch (UnknownDomainObjectException ignore) {
+            return project.getRootProject().getGradle().includedBuild(projectComponentIdentifier.getBuild().getName());
+        } catch (UnknownDomainObjectException e) {
             return null;
         }
     }
 
+    public static Project includedBuildProject(IncludedBuildInternal includedBuild,
+            final ProjectComponentIdentifier componentIdentifier) {
+        return includedBuild.getTarget().getMutableModel().getRootProject().findProject(
+                componentIdentifier.getProjectPath());
```

Comment:
Is there a way to get hold of a project using non-internal API?

Reply:
I was also kind of hesitant to use this, as relaying on Internal implementation details usually has ‏consequences, but haven't found a different way. I believe there's no other way, as the current Gradle project (under build) is by definition unrelated to the included build, so getting a reference to the other Gradle project can't be done through the current Gradle project object graphs.